src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker_ResolveReferences.cs

```diff
@@ -157,11 +158,15 @@ private ImmutableArray<MetadataReference> GetMetadataReferences()
                 {
                     var builder = ImmutableArray.CreateBuilder<MetadataReference>();
 
+                    // used to eliminate duplicates
+                    var _ = PooledHashSet<MetadataReference>.GetInstance(out var set);
+
                     for (var index = 0; index < _metadataReferences.Length; index++)
                     {
-                        if (!_indicesToRemove.Contains(index))
+                        var reference = _metadataReferences[index];
+                        if (!_indicesToRemove.Contains(index) && set.Add(reference))
```

Comment:
Does _indicesToRemove have to be updated if we skipped one? I don't recall what that magic thing does, but if some later bit expects a 1:1 mapping this is now breaking that.

Reply:
(Or does _indicesToRemove have to be the computation that figures out the duplicates?)

Reply:
`_indicesToRemote` is set of indices in the original array that should be removed. These indices are unrelated to the items in the builder.